src/iris/api.py

```diff
@@ -732,13 +735,18 @@ def process_resource(self, req, resp, resource, params):  # pragma: no cover
         # Ignore HMAC requirements for custom webhooks
         if req.env['PATH_INFO'].startswith('/v0/webhooks/'):
             app_name = req.get_param('application', required=True)
+
             app = cache.applications.get(app_name)
             if not app:
                 raise HTTPUnauthorized('Authentication failure',
                                        'Application not found', [])
 
             req.context['app'] = app
 
+            # optionally used and validated based on webhook impl
```

Comment:
Instead of finding the plan in the AuthMiddleware, do so in the rackspace `on_post()` method. Falcon passes the same `req` object to `on_post()`, so the implementation should be the same, but it's cleaner to group this with the other rackspace code.